### PR TITLE
Add Gutenberg Whitelisting/blacklisting

### PR DESCRIPTION
**Pull request by Firestorm980**
### Description of the Change

Updates Gutenberg best practices to include information on whitelisting and blacklisting of blocks as appropriate to a project. Gives some circumstances on when this may be desired or needed.

### Alternate Designs

None.

### Benefits

Additional information available for engineers, PMs, etc.

### Possible Drawbacks

May not include all scenarios.

### Checklist:

- [x] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

Request from https://docs.google.com/a/get10up.com/document/d/1Cf4DESCWyRSIJ0NM7xaJgf00aLcBnXPIM262wCLhKOo/edit?disco=AAAADokYgRw

### CODE REVIEW

**Review by timwright12 — APPROVED** [review]

File: _includes/markdown/React.md [hunk 1]
```diff
@@ -151,3 +151,18 @@ The presence of Element is why you don't see React directly imported into Gutenb
...
 
 As with any evolving feature, it is important to frequently check the documentation for new additions and updates.
+
+### Whitelisting / Blacklisting
+
+Gutenberg has the ability to whitelist or blacklist certain blocks. This can be found under the [Block Filters](https://developer.wordpress.org/block-editor/developers/filters/block-filters/) page of the official Gutenberg documentation. Specifically, under the heading [Removing Blocks](https://developer.wordpress.org/block-editor/developers/filters/block-filters/#removing-blocks). Details of how to filter via JavaScript or PHP can be found there.
+
+Also note that this method is different than the `allowedBlocks` prop on `<InnerBlocks>` when used in a custom block. Whitelisting or blacklisting here applies to all blocks on a higher level within the editor.
+
+There may be a need to remove or restrict access to blocks. This could be for a variety of reasons, but typical scenarios could be:
```
Comment (timwright12) [inline _includes/markdown/React.md:171]: @Firestorm980 can we flesh this recommendation out a bit more? Maybe something more around "this is when you whitelist" vs. "this is when you blacklist"?

**Review by timwright12 — CHANGES_REQUESTED** [review]

**Comment by timwright12** [comment]
@Firestorm980 just checking in on this

**Comment by Firestorm980** [comment]
@timwright12 I have some time here, should be able to update. I'm guessing in the majority of cases, we'd reach for whitelisting in projects versus a blacklist, since a whitelist would likely have to be maintained less over time and allows for better control/security.

File: _includes/markdown/React.md (hunk 1 above)
Comment (Firestorm980) [reply _includes/markdown/React.md:171]: Updated

**Review by timwright12 — APPROVED** [review]

**Comment by samikeijonen** [comment]
Suggestion about wording. Can we change wording about whitelist and blacklist to allowed blocks and denied blocks. Or something like that.

I'm not sure what wording WP Core uses but I'd like us to move away from those old naming conventions for better inclusiveness.

**Comment by Firestorm980** [comment]
> Suggestion about wording. Can we change wording about whitelist and blacklist to allowed blocks and denied blocks. Or something like that.
> 
> I'm not sure what wording WP Core uses but I'd like us to move away from those old naming conventions for better inclusiveness.

@samikeijonen added some changes to remove those terms. Please give it another review when you get the opportunity.

**Comment by timwright12** [comment]
@brentvr can you take a look at this?

**Comment by jeffpaul** [comment]
@brentvr any updates on your review?

**Comment by tlovett1** [comment]
Looks solid. Thanks @Firestorm980 !

**Pull request merged**